cmd/compare.go

```diff
@@ -94,85 +104,145 @@ func compare(provider string, oldCommit string, newCommit string) error {
 		for propName, prop := range res.Properties {
 			newProp, ok := newRes.Properties[propName]
 			if !ok {
-				violations = append(violations, fmt.Sprintf("Resource %q missing output %q", resName, propName))
+				violation("missing output %q", propName)
 				continue
 			}
 
 			vs := validateTypes(&prop.TypeSpec, &newProp.TypeSpec, fmt.Sprintf("Resource %q output %q", resName, propName))
 			violations = append(violations, vs...)
 		}
+
+		oldRequiredInputs := newSetFromList(res.RequiredInputs)
+		for _, input := range newRes.RequiredInputs {
+			if !oldRequiredInputs.Has(input) {
+				violation("added new required input %q", input)
```

Comment:
```suggestion
				violation("input %q has changed to Required", input)
```

Reply:
To me, the wording was ambiguous, i.e. folks reading the schema diff might skip over the "required" part and think the whole resource/type/etc is new.